include/NAS2D/Renderer/Color.h

```diff
@@ -21,6 +21,23 @@ namespace NAS2D {
 class Color_4ub
 {
 public:
+	static const Color_4ub Black;
+	static const Color_4ub Blue;
+	static const Color_4ub Bright_green;
+	static const Color_4ub Cyan;
+	static const Color_4ub Green;
+	static const Color_4ub Grey;
+	static const Color_4ub Magenta;
+	static const Color_4ub Navy;
+	static const Color_4ub Orange;
+	static const Color_4ub Red;
+	static const Color_4ub Silver;
+	static const Color_4ub White;
+	static const Color_4ub Yellow;
+	static const Color_4ub Normal;
+	static const Color_4ub NormalZ;
+	static const Color_4ub NoAlpha;
```

Comment:
Can probably just declare these as `static constexpr`, and then initialize them in the header. A `constexpr` variable is also implicitly an `inline` variable.

Reference: [How do inline variables work?](https://stackoverflow.com/questions/38043442/how-do-inline-variables-work)

Reply:
> 
> 
> Can probably just declare these as `static constexpr`, and then initialize them in the header. A `constexpr` variable is also implicitly an `inline` variable.
> 
> Reference: [How do inline variables work?](https://stackoverflow.com/questions/38043442/how-do-inline-variables-work)

I keep getting either `C2078: Too many initializers` when I use `static const Color_4ub Black{0,0,0,255}` or `C2059: syntax error 'constant'` when I use `static const Color_4ub Black(0,0,0,255)` and `expression must have a constant value` or `expected a type specifier` when each is used with `constexpr` respectively.

Reply:
Hmm, I played around with this a bit. I managed to get something working, but not with the constants inside of the `Color_4ub` class. In hindsight, it does seem a little bit strange to have them there. The constants are static instances contained in their own class type. Their own class type will necessarily be incomplete until after the scope has ended, and class scope can't be re-opened for addition.

Instead, to use `constexpr`, I had to declare them after the close of the class. I also needed to declare the constructor as `constexpr`. That requires all initialization to be done through the initializer list (which is already true here). It also requires the full constructor definition (not just the declaration) be visible before defining the objects that use the constructor. If the `constexpr` variables are defined in the header, then the constructor definition also needs to be contained in the header.

Given the design considerations, I'm tempted to merge this as is. If we wish to investigate using `constexpr`, perhaps that would be best as its own issue. Will revisit tomorrow, unless Leeor wants to make a decision meanwhile.

Reply:
Was going to say, constexpr should be a separate issue. This will suffice for now. Thanks!